torch/_decomp/decompositions.py

```diff
@@ -1962,6 +1962,10 @@ def uniform(
         device=x.device,
     )
 
+@register_decomposition(aten.geometric_)
+def geometric(self, p):
+    return self.copy_(torch.floor(torch.log(1 - torch.rand_like(self)) / torch.log(torch.tensor(1 - p))) + 1)
```

Comment:
Thanks @jgong5, actually `1-U` just transforms `U` in range `[0, 1)` (which is the range of `torch.rand`, `torch.rand_like` ) to `(0, 1]` to avoid `log(0)`.  

`log(1-U)` for `U `in `[0, 1)`
`log(U)` for `U` in `(0, 1]` or `(0, 1)`.

And I think the formula https://pytorch.org/docs/stable/generated/torch.Tensor.geometric_.html here is incorrect.

Reply:
Sure, I understood log(U) and log(1-U) part but log(p) vs. log(1-p) part is different. You mean the doc is incorrect. Shall we fix that too?

Reply:
Do you mean `log(q) = log(1-p)`? `q = 1 - p`

Yes, I think the doc is incorrect (pretty sure..) https://github.com/pytorch/pytorch/pull/92049

Reply:
Oh, you meant `log(p)` if the doc were correct

Reply:
The formula in the docs is certainly non-standard. See https://en.wikipedia.org/wiki/Geometric_distribution. We should improve the docs with a description of what does this model (in this case, probability until the first miss).

For the formula I'd suggest two things. First: How is it implemented in core? And second, consider using `torch.log1p` for better stability around 0.

Reply:
> The formula in the docs is certainly non-standard. See https://en.wikipedia.org/wiki/Geometric_distribution. We should improve the docs with a description of what does this model (in this case, probability until the first miss). How is it implemented in core?

The [core ](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/TransformationHelper.h#L149-L153)implements a standard geometric distribution, the first one described in the linked wiki.  
`P{X=i} = p * q^(i-1), i >= 1, where q = 1 -p` 
`p: probability of success`
`q: probability of failure`
`P{X=i}: probability of first success on the ith trial`

Note that, the inplace geometric https://pytorch.org/docs/stable/generated/torch.Tensor.geometric_.html models the first one described in the linked wiki while `torch.distributions.geometric.Geometric` https://pytorch.org/docs/stable/distributions.html?highlight=geometric#torch.distributions.geometric.Geometric models the second one described in the linked wiki. 

>  And second, consider using torch.log1p for better stability around 0

Thanks @lezcano !

Reply:
Sorry, had a closer look at what `torch.log1p` does. And I don't this `torch.log1p` is suitable to just add `1+input`.

i.e., `floor( log(2-U) / log(2-p) ) + 1` is certainly not the same as  `floor( log(1-U) / log(1-p) ) + 1`

Reply:
See how it's implemented here:
https://pytorch.org/docs/stable/_modules/torch/distributions/geometric.html#Geometric

Reply:
Thanks @lezcano, done, makes sense.